test/e2e/metamask-ui.spec.js

```diff
@@ -1435,19 +1445,21 @@ describe('MetaMask', function () {
     });
 
     it('finds the transaction in the transactions list', async function () {
-      await driver.wait(async () => {
-        const confirmedTxes = await driver.findElements(
-          '.transaction-list__completed-transactions .transaction-list-item',
-        );
-        return confirmedTxes.length === 4;
-      }, 10000);
+      await driver.waitForSelector({
+        // Select the heading of the first transaction list item in the
+        // completed transaction list with text matching Send TST
+        css:
+          '.transaction-list__completed-transactions .transaction-list-item:first-child .list-item__heading',
+        text: 'Send TST',
+      });
 
-      const txValues = await driver.findElements(
-        '.transaction-list-item__primary-currency',
+      await driver.waitForSelector(
+        {
+          css: '.transaction-list-item__primary-currency',
+          text: '-1.5 TST',
+        },
+        { timeout: 10000 },
```

Comment:
Why so long of a wait here? Is this the same transaction that's being waited for on the line above this? If so I'd expect it to be fairly quick to locate.